vowpalwabbit/core/src/reductions/details/automl_impl.h

```diff
@@ -90,42 +92,40 @@ struct config_oracle
   std::string _interaction_type;
   const std::string _oracle_type;
 
-  // insert_config(..)
-  std::priority_queue<std::pair<float, uint64_t>>& index_queue;
+  // Maybe not needed with oracle, maps priority to config index, unused configs
+  std::priority_queue<std::pair<float, uint64_t>> index_queue;
+  // Stores all configs in consideration
+  std::vector<exclusion_config> configs;
   std::map<namespace_index, uint64_t>& ns_counter;
-  std::vector<exclusion_config>& configs;
 
   priority_func* calc_priority;
   const uint64_t global_lease;
   uint64_t valid_config_size = 0;
   oracle_impl _impl;
 
-  config_oracle(uint64_t global_lease, priority_func* calc_priority,
-      std::priority_queue<std::pair<float, uint64_t>>& index_queue, std::map<namespace_index, uint64_t>& ns_counter,
-      std::vector<exclusion_config>& configs, const std::string& interaction_type, const std::string& oracle_type,
-      std::shared_ptr<VW::rand_state>& rand_state);
+  config_oracle(uint64_t global_lease, priority_func* calc_priority, std::map<namespace_index, uint64_t>& ns_counter,
+      const std::string& interaction_type, const std::string& oracle_type, std::shared_ptr<VW::rand_state>& rand_state);
 
-  void do_work(std::vector<std::pair<aml_estimator, estimator_config>>& estimators, const uint64_t current_champ);
+  void do_work(estimator_vec_t& estimators, const uint64_t current_champ);
   void insert_config(std::set<std::vector<namespace_index>>&& new_exclusions, bool allow_dups = false);
   bool repopulate_index_queue();
+  void insert_qcolcol();
 };
 
 struct oracle_rand_impl
 {
   std::shared_ptr<VW::rand_state> random_state;
   oracle_rand_impl(std::shared_ptr<VW::rand_state> random_state) : random_state(std::move(random_state)) {}
-  void do_work(config_oracle<oracle_rand_impl>* config_oracle,
-      std::vector<std::pair<aml_estimator, estimator_config>>& estimators, const uint64_t current_champ);
+  void do_work(
+      config_oracle<oracle_rand_impl>* config_oracle, estimator_vec_t& estimators, const uint64_t current_champ);
 };
 struct one_diff_impl
 {
-  void do_work(config_oracle<one_diff_impl>* config_oracle,
-      std::vector<std::pair<aml_estimator, estimator_config>>& estimators, const uint64_t current_champ);
+  void do_work(config_oracle<one_diff_impl>* config_oracle, estimator_vec_t& estimators, const uint64_t current_champ);
 };
 struct champdupe_impl
 {
-  void do_work(config_oracle<champdupe_impl>* config_oracle,
-      std::vector<std::pair<aml_estimator, estimator_config>>& estimators, const uint64_t current_champ);
+  void do_work(config_oracle<champdupe_impl>* config_oracle, estimator_vec_t& estimators, const uint64_t current_champ);
```

Comment:
Could we name the `do_work` functions with something more specific to what they do

Reply:
done